frontend/src/common/queries/collections.ts

```diff
@@ -258,14 +315,21 @@ const editCollection = async function ({
   return result;
 };
 
-export function useEditCollection() {
+export function useEditCollection(collectionID?: Collection["id"]) {
   const queryCache = useQueryCache();
 
+  const { data: collection } = useCollection({
+    id: collectionID,
+    visibility: VISIBILITY_TYPE.PRIVATE,
+  });
+
   return useMutation(editCollection, {
-    onSuccess: (collection: Collection) => {
-      return queryCache.setQueryData(
-        [USE_COLLECTION, collection.id, collection.visibility],
-        collection
+    onSuccess: (newCollection) => {
+      queryCache.setQueryData(
+        [USE_COLLECTION, collectionID, VISIBILITY_TYPE.PRIVATE],
+        () => {
+          return { ...collection, ...newCollection };
```

Comment:
ah can you remind me why we need to spread `...collection, ...newCollection` here instead of just using `newCollection`? Thank you!

Reply:
`collection` holds some locally calculated attributes that `newCollection` won't hold